src/components/sections/LocationCardFactory.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom'; // Import Link from React Router
+import Parse from 'parse/dist/parse.min.js';
+import { List } from 'antd';
+import Card from 'react-bootstrap/Card';
+import Container from 'react-bootstrap/Container';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faHeart } from '@fortawesome/free-solid-svg-icons';
+import '../../styles/index.css';
+
+export const LocationCardFactory = () => {
+    const [readResults, setReadLocations] = useState([]);
+    const [loading, setLoading] = useState(true);
+
+    const readLocations = async function () {
+        const parseQuery = new Parse.Query('Location');
+        parseQuery.include('Country');
+        try {
+            let locations = await parseQuery.find();
+            setReadLocations(locations);
+            setLoading(false);
+            return true;
+        } catch (error) {
+            alert(`Error! ${error.message}`);
+            setLoading(false);
+            return false;
+        }
+    };
+
+    useEffect(() => {
+        readLocations();
+    }, []);
+
+    return (
+        <div>
+            <div className="location-container">
+                <div>
+                    {loading && <div>Loading...</div>}
+                    {!loading && readResults !== null &&
+                        readResults !== undefined &&
+                        readResults.length > 0 && (
+                            <List
+                                dataSource={[...readResults].reverse()}
+                                renderItem={(item) => (
+                                    <List.Item className="card-items">
```

Comment:
this would be a good candidate for a component? 
then the html markup would not be so deeply nested/